### PR TITLE
[LW-9482] Fix/cip30 sign data

### LINKED ISSUE #803: Cannot sign message to get signature
I used signMessage in useCardano function to get the wallet's signature. but it reported missing password error
image

### PR DESCRIPTION
**Pull request by mchappell**
# Checklist

- [ ] JIRA - \<link>
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

`signBlob` method from keyAgent is now also wrapped, so that the auto-redirect to sign success state can be removed, this was prematurely removing password from memory which caused the blob not to sign

## Testing

Execute at example.com in console

```
const api = await window.cardano.lace.enable();
const ownAddress = await api.getUsedAddresses();
const signedData = await api.signData(
        ownAddress[0],
        '66697865642074686520627567'
      );
```
## Screenshots

image
image

Fixes https://github.com/input-output-hk/lace/issues/803

### CODE REVIEW

**Review by lucas-barros — APPROVED** [review]

File: packages/cardano/src/wallet/lib/cardano-wallet.ts [hunk 1]
```diff
@@ -127,31 +127,45 @@ export const createCardanoWalletsByChain = async (
   return { wallet, keyAgent: activeKeyAgent, keyAgentsByChain };
 };
 
+const keyagentSigningCallback = (signCallback?: (result: boolean) => void, response?: boolean) => {
```
Comment (pczeglik-iohk) [inline packages/cardano/src/wallet/lib/cardano-wallet.ts:130]: Can we slightly improve the name of this function:`keyAgentSigningCallback`?

**Review by pczeglik-iohk — APPROVED** [review]
I spot one small think, but it's not a deal breaker :)

File: packages/cardano/src/wallet/lib/cardano-wallet.ts (hunk 1 above)
Comment (mchappell) [reply packages/cardano/src/wallet/lib/cardano-wallet.ts:130]: `executeSigningCallbackIfExists` ?

**Review by ljagiela — APPROVED** [review]

**Review by danielmain — APPROVED** [review]

File: packages/cardano/src/wallet/lib/cardano-wallet.ts (hunk 1 above)
Comment (pczeglik-iohk) [reply packages/cardano/src/wallet/lib/cardano-wallet.ts:130]: Sounds good to me :)

**Review by wklos-iohk — APPROVED** [review]

**Pull request merged**